middleware.ml

```diff
@@ -101,7 +101,8 @@ let redirect_to_dashboard reqd ?(msg = "") () =
   Httpaf.Reqd.respond_with_string reqd response msg;
   Lwt.return_unit
 
-let http_response ~title ?(header_list = []) ?(data = "") reqd http_status =
+let http_response ~title ?(header_list = []) ?(data = `String "") reqd
```

Comment:
What is the reasoning behind this change, putting `data` as some json object instead of a string?

I'm asking since this change adds a lot of changes to this PR, and I'm not able to figure out why that needs to change?

I was able to find this code:
```Diff
                        Middleware.http_response reqd ~header_list
                          ~title:"Success"
                          ~data:
                            (Yojson.Basic.to_string
                               (User_model.user_to_json user))
                          ~data:(User_model.user_to_json user)
                          `OK
```

But am still unsure why this json object is now necessary.

Reply:
So the reason why `data` is now a json object is because we use it for both responses to the API and also responses to the UI. When we format data as a string, the api responses also return with data as a string. When we make data an object, we can have the API responses return correctly. This is an example of the response when creating a new token via the API.

Without `data` as a json object this is what the results come as:

```json
{
    "status": 200,
    "title": "Success",
    "success": true,
    "data": {
    "status": 200,
    "title": "Success",
    "success": true,
    "data": "{\"token_type\":\"Bearer\",\"value\":\"7888febc-fb8c-4648-acb8-e449ba27ee88\",\"expires_in\":3600,\"created_at\":\"2024-12-09 11:46:01-00:00\",\"last_access\":\"2024-12-09 11:46:01-00:00\",\"name\":\"api_test_token\",\"usage_count\":0}"
}
}
```

with data as a json object, we can have this, and then for results of the UI such as in notifications we can do it as a string:
```json
{
    "status": 200,
    "title": "Success",
    "success": true,
    "data": {
        "token_type": "Bearer",
        "value": "6b8fe160-1dbd-4193-a5f8-8459d2c5d59a",
        "expires_in": 3600,
        "created_at": "2024-12-09 11:39:59-00:00",
        "last_access": "2024-12-09 11:39:59-00:00",
        "name": "api_test_token",
        "usage_count": 0
    }
}
```